contracts/bonding/BondingVotes.sol

```diff
@@ -0,0 +1,558 @@
+     */
+    constructor(address _controller) Manager(_controller) {}
+
+    // IVotes interface implementation.
+    // These should not access any storage directly but proxy to the bonding state functions.
+
+    /**
+     * @notice Returns the name of the virtual token implemented by this.
+     */
+    function name() external pure returns (string memory) {
+        return "Livepeer Voting Power";
+    }
+
+    /**
+     * @notice Returns the symbol of the token underlying the voting power.
+     */
+    function symbol() external pure returns (string memory) {
+        return "vLPT";
+    }
+
+    /**
+     * @notice Returns the decimals places of the token underlying the voting.
+     */
+    function decimals() external pure returns (uint8) {
+        return 18;
+    }
+
+    /**
+     * @notice Clock is set to match the current round, which is the checkpointing
+     *  method implemented here.
+     */
+    function clock() public view returns (uint48) {
+        return SafeCast.toUint48(roundsManager().currentRound());
+    }
+
+    /**
+     * @notice Machine-readable description of the clock as specified in EIP-6372.
+     */
+    // solhint-disable-next-line func-name-mixedcase
+    function CLOCK_MODE() external pure returns (string memory) {
+        return "mode=livepeer_round";
+    }
+
+    /**
+     * @notice Returns the current amount of votes that `_account` has.
+     * @dev Keep in mind that since this function should return the votes at the end of the current round, we need to
+     * fetch the bonding state at the next round instead. That because the bonding state reflects the active stake in
+     * the current round, which is the snapshotted stake from the end of the previous round.
+     */
+    function getVotes(address _account) external view returns (uint256) {
+        (uint256 amount, ) = getBondingStateAt(_account, clock() + 1);
+        return amount;
+    }
+
+    /**
+     * @notice Returns the amount of votes that `_account` had at a specific moment in the past. If the `clock()` is
+     * configured to use block numbers, this will return the value at the end of the corresponding block.
+     * @dev Keep in mind that since this function should return the votes at the end of the _round (or timepoint in OZ
+     * terms), we need to fetch the bonding state at the next round instead. That because the bonding state reflects the
+     * active stake in the current round, which is the snapshotted stake from the end of the previous round.
+     */
+    function getPastVotes(address _account, uint256 _round) external view onlyPastRounds(_round) returns (uint256) {
+        (uint256 amount, ) = getBondingStateAt(_account, _round + 1);
+        return amount;
+    }
+
+    /**
+     * @notice Returns the current total supply of votes available.
+     * @dev This value is the sum of all *active* stake, which is not necessarily the sum of all voting power.
```

Comment:
After thinking on this a bit, this feels weird.

I would expect that totalSupply() == sum(voting power). While not having this invariant doesn't seem like it would break anything, not having this invariant in place feels like smell...if I query totalSupply() and then index all votes for a proposal I would expect to be able to calculate the % participation in a vote as sum(voting power indexed) / totalSupply(), but that would not be wrong because the denominator is incorrect.

I know the current offchain poll tallying logic has the same problem, but if the lift to correct this problem is small (keeping track of an additional state variable in checkpoints to determine if an address was either an active transcoder or delegated to an active transcoder as of the timepoint?) to I would lean towards fixing it to start off on solid footing with proper invariants in place for onchain voting.

Reply:
Discussed offline addressing this separately from this PR

Reply:
FTR we discussed offline and concluded the current behavior is reasonable given the current audit contest timelines. Might followup with another PR changing this behavior if there's time, but focusing instead on documenting the current behavior and contest for now.